src/main/java/org/elasticsearch/search/aggregations/bucket/terms/support/IncludeExclude.java

```diff
@@ -89,12 +93,20 @@ private void addReject(long val) {
      * @param excludeValues   The terms to be excluded
      *                  (may only be {@code null} if one of the other arguments is none-null.
      */
-    public IncludeExclude(Pattern include, Pattern exclude, Set<BytesRef> includeValues, Set<BytesRef> excludeValues) {
-        assert includeValues != null || include != null || 
+    public IncludeExclude(RegExp include, RegExp exclude, Set<BytesRef> includeValues, Set<BytesRef> excludeValues) {
+        assert includeValues != null || include != null ||
                 exclude != null || excludeValues != null : "includes & excludes cannot both be null"; // otherwise IncludeExclude object should be null
-        this.include = include != null ? include.matcher("") : null;
-        this.exclude = exclude != null ? exclude.matcher("") : null;
-        hasRegexTest = include != null || exclude != null;
+        if (include != null) {
+            // determinize first so that we don't determinize twice because of
+            // the ByteRunAutomaton and the CompiledAutomaton
+            final Automaton a = Operations.determinize(include.toAutomaton(), Operations.DEFAULT_MAX_DETERMINIZED_STATES);
```

Comment:
Why do we need both? Is it because there are so many things going on in this file? I don't understand why we wouldnt just need the CompiledAutomaton for the terms.intersect operation, why do we need a ByteRunAutomaton too?

Having both seems silly anyway, but if we must do it, try to assign the ByteRunAutomaton from the CompiledAutomaton. The majority of the time it will be non-null:

    /** 
     * Matcher for quickly determining if a byte[] is accepted.
     * only valid for {@link AUTOMATON_TYPE#NORMAL}.
     */
    public final ByteRunAutomaton runAutomaton;